src/float.cr

```diff
@@ -148,8 +148,17 @@ struct Float32
     Printer.print(self, io)
   end
 
+  private def hash_normalize
+    t = to_i32
+    t == self ? t : unsafe_as(Int32)
```

Comment:
```crystal
0.5_f32.hash == 0.5_f64.hash # => true
0.75_f32.hash == 0.75_f64.hash # => true
4.1_f32.hash == 4.1_f64.hash # => false - because values differ.
```